### ISSUE TITLE
[BUG] [Java Service]Source and translation are null when requesting inexistent key without source parameter in string-based GET API.

### ISSUE THREAD

**Issue by lyiyu66** [vmware/singleton#1453]
**Describe the bug**
commit: c2c449489db2d2608f04cdf63baa55f72ac1cafc
Source and translation returns null when requesting inexistent key without source parameter in string-based GET API.
It occurs in v1 and v2 API.
v1: /i18n/api/v1/translation/product/{productName}/key/{key}
v2: /i18n/api/v2/translation/products/{productName}/versions/{version}/locales/{locale}/components/{component}/keys/{key}

**To Reproduce**
1. Go to get a source API:
/i18n/api/v2/translation/products/{productName}/versions/{version}/locales/{locale}/components/{component}/keys/{key}
2. Add below parameters to API request: (make sure requested product(i.e. VMwareVIP2018) is in product supported list)
https://localhost:8090/i18n/api/v2/translation/products/VMwareVIP2018/versions/1.1.0/locales/zh-hans/components/default/keys/Partner_Name1?collectSource=true&pseudo=false
3. Get the response as below:
data/source: null
data/translation: null
```
{
  "response": {
    "code": 200,
    "message": "OK",
    "serverTime": ""
  },
  "signature": "",
  "data": {
    "productName": "VMwareVIP2018",
    "version": "1.1.0",
    "dataOrigin": "",
    "pseudo": false,
    "machineTranslation": false,
    "source": null,
    "translation": null,
    "locale": "zh-Hans",
    "key": "Partner_Name1",
    "component": "default",
    "status": "The translation is not found , English not found, return the recieved source as translation."
  }
}
```

**Expected behavior**
data/source should be empty as "".
```
"source": "",
"translation": ""
```

**Comment by lyiyu66** [comment]
Same issue about translation is null in API: 
POST
/i18n/api/v2/translation/products/{productName}/versions/{version}/locales/{locale}/components/{component}/keys/{key}

https://localhost:8090/i18n/api/v2/translation/products/VMwareVIP2018/versions/1.1.0/locales/en/components/default/keys/Partner_1?checkTranslationStatus=false&collectSource=false&machineTranslation=false&pseudo=false

```
{
  "response": {
    "code": 200,
    "message": "OK",
    "serverTime": ""
  },
  "signature": "",
  "data": {
    "productName": "VMwareVIP2018",
    "version": "1.1.0",
    "dataOrigin": "",
    "pseudo": false,
    "machineTranslation": false,
    "source": "",
    "translation": null,
    "locale": "en",
    "key": "Partner_1",
    "component": "default",
    "status": "The translation is found and returned, but the according en string is not matched to source."
  }
}
```

**Comment by InaSong** [comment]
Assign to Tiger for first round of triage.

**Comment by lyiyu66** [comment]
The issue is fixed with commit 93ffa26a7c0616ff7fee3eb25318342d1dc8975b
